sstables/super_sstable_reader.go

```diff
@@ -108,8 +109,11 @@ func ScanReduceLatestWins(key []byte, values [][]byte, context []int) ([]byte, [
 			maxCtxIndex = i
 		}
 	}
-
-	return key, values[maxCtxIndex]
+	val := values[maxCtxIndex]
+	if len(val) == 0 {
```

Comment:
value is not nil for thumbstone as it is said in memstore.go

	// This includes tombstoned keys and writes their values as nil.
	FlushWithTombstones(opts ...sstables.WriterOption) error

Reply:
could you please confirm my understanding ? Or not ?

Reply:
Sorry, I thought this was a comment for you. Your understanding is correct, as far as I remember it.

Reply:
ok, so the comment in memstore.go is false ? How can we save empty value ?

Reply:
hmm. So when we tombstone through the memstore, we create a struct value that has NIL bytes when the key previously existed:
https://github.com/thomasjungblut/go-sstables/blob/main/memstore/memstore.go#L147-L160
When it didn't exist, it creates an empty slice as for the var declaration...

Then there's also what we actually write to the file. Let me quickly cobbled together a testcase and fix the stuff around this.

Reply:
apologies @sebheitzmann - this turned out to be across the whole stack :/ 
I'm starting to create a new recordio version here https://github.com/thomasjungblut/go-sstables/pull/34
Gotta spend some more time to write regression tests, not all compatibility tests are running correctly either.

Reply:
Thank you.